analyzers/src/SonarAnalyzer.CSharp/Rules/MemberInitializerRedundant.cs

```diff
@@ -63,168 +61,178 @@ public sealed class MemberInitializerRedundant : SonarDiagnosticAnalyzer
                         return;
                     }
 
-                    var constructorSymbols = members.OfType<IMethodSymbol>().Where(IsExplicitlyDefinedConstructor).ToList();
-                    ReportRedundantInitializers(c,
-                        declaration,
-                        constructorSymbols,
-                        x => !x.Modifiers.Any(IsStaticOrConst),
-                        x => !x.Modifiers.Any(IsStaticOrConst));
-
-                    var staticInitializerSymbols = members.OfType<IMethodSymbol>().Where(IsStaticInitializer).ToList();
```

Comment:
[refactoring help] this has been moved in the `InitializationVerifier.Report()` method